org.opent2t.sample.hub.superpopular/com.insteon.hub/js/thingTranslator.js

```diff
@@ -0,0 +1,300 @@
+                            
+                            return Promise.resolve(device);
+                        }
+                        return Promise.resolve(undefined);
+                    });
+
+                promises.push(promise);
+            });
+
+            return Promise.all(promises).then((results) => {
+                for (var i = 0; i < results.length ; i++) {
+                    if(results[i] !== undefined){
+                        filteredDevices.push(results[i]);
+                    }
+                }
+
+                toReturn.devices = filteredDevices;
+                return Promise.resolve(toReturn);
+            });
+        });
+    }
+
+    /**
+     * Get the name of the hub.  Ties to the n property from oic.core
+     */
+    getN() {
+        return this._name;
+    }
+
+    /**
+     * Gets device details (all fields), response formatted per http://docs.insteon.apiary.io/#reference/devices
+     */
+    getDeviceDetailsAsync(deviceId) {
+
+        // Make the async request
+        return this._makeRequest(this._devicesPath + '/' + deviceId, 'GET')
+            .then((details) => {
+
+                var device = details;
+
+                var postPaylaod =
+                {
+                    command: 'get_status',
+                    device_id: deviceId
+                };
+
+                return this._makeRequest(this._commandPath, 'POST', JSON.stringify(postPaylaod))
+                    .then((response) => {                       
+                        return this._getCommandResponse(response.id, 3) // get device status
+                            .then((deviceStatus) => {
+
+                                if (deviceStatus !== undefined && deviceStatus.status == 'succeeded') {   
+                                    device['Level'] = deviceStatus.response.level;
+                                    if (deviceStatus.response.level == 0) {
+                                        device['Power'] = 'off';
+                                    } else {
+                                        device['Power'] = 'on';
+                                    }
+                                }
+                                return Promise.resolve(device);
+                            });
+                    });
+            });
+    }
+
+    /**
+     * Puts device details (all fields) payload formatted per http://docs.insteon.apiary.io/#reference/devices
+    */
+    putDeviceDetailsAsync(deviceId, putPayload) {
+
+        // build request path and body
+        var statusChanges = { device_id: deviceId };
+        var nonStatusChanges = {};
+        
+        for(var item in putPayload){
+            if(this._isDeviceStatus(item)){
+                statusChanges[item] = putPayload[item];
+            } else {
+                nonStatusChanges[item] = putPayload[item];
+            }
+        }
+
+        var translatorCopy = new Translator(this._accessToken);
```

Comment:
I am not sure the reason, but when I call _getCommandResponse at line 143,  'this ' becomes 'undefined', which blocks me from calling _getCommandResponse. I made a translator copy as a workaround.

---
In reply to: [82257940](https://github.com/openT2T/translators/pull/107#discussion_r82257940) [](ancestors = 82257940)